biz.aQute.remote/src/aQute/remote/agent/AgentServer.java

```diff
@@ -627,9 +629,10 @@ public void frameworkEvent(FrameworkEvent event) {
 				refresh.await();
 				return;
 			}
-		} catch (Exception e) {
-			ServiceReference<org.osgi.service.packageadmin.PackageAdmin> ref = context
-					.getServiceReference(org.osgi.service.packageadmin.PackageAdmin.class);
+		} catch (Exception | NoSuchMethodError e) {
+			@SuppressWarnings("unchecked")
+			ServiceReference<org.osgi.service.packageadmin.PackageAdmin> ref = (ServiceReference<PackageAdmin>) context
```

Comment:
Use fully qualified name for PackageAdmin to avoid need to import and thus avoid unsupressable deprecation warnings.